templates/htan-synapse-bucket.yaml

```diff
@@ -70,12 +70,22 @@ Resources:
             AllowedOrigins: ['*']
             AllowedMethods: [GET, POST, PUT, HEAD]
             MaxAge: 3000
-      NotificationConfiguration:
-        LambdaConfigurations:
-          - Event: "s3:ObjectCreated:*"
-            Function: !Ref S3SynapseSyncFunctionArn
-          - Event: "s3:ObjectRemoved:*"
-            Function: !Ref S3SynapseSyncFunctionArn
+       # LambdaBucketPermission must be setup before NotificationConfiguration
+       # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-notificationconfig.html
+#      NotificationConfiguration:
+#        LambdaConfigurations:
+#          - Event: "s3:ObjectCreated:*"
+#            Function: !Ref S3SynapseSyncFunctionArn
+#          - Event: "s3:ObjectRemoved:*"
+#            Function: !Ref S3SynapseSyncFunctionArn
+  LambdaBucketPermission:
```

Comment:
Adding NotificationConfiguration would fail without LambdaBucketPermission.

I think this can replace the `Sid: "LambdaAccess"` section below.  What do you think @clarisse-lau ?

Reply:
Does not this one give the right to the bucket to invoke the function, and LambdaAccess give the function role the right to get/putObject in the bucket?

Reply:
Yeah, this is the potential issue I was wondering about in my [comment](https://github.com/Sage-Bionetworks/scicomp-provisioner/pull/374#issuecomment-681008691) yesterday. I think we should try this out. But agree with @xschildw , I think we still need the `Sid: "LambdaAccess"` section